day5_2018.py

```diff
@@ -0,0 +1,79 @@
+input5 = [open("input5.txt").read()]
+
+
+def get_separate_letters(inputs):
+    new_inputs = []
+    for line in inputs:
+        for letter in line:
+            new_inputs.append(letter)
+    return new_inputs
+
+
+def pop_letters_from_inputs(inputs, letter_idx):
+    inputs.pop(letter_idx)
+    inputs.pop(letter_idx)
+
+
+def get_all_possible_reactions(inputs):
+    for letter_idx, first_letter in enumerate(inputs):
+        if letter_idx == len(inputs) - 1:
+            break
+        for second_letter in inputs[letter_idx + 1]:
+            if first_letter.isupper():
+                if second_letter == first_letter.lower():
+                    pop_letters_from_inputs(inputs, letter_idx)
+            if first_letter.islower():
+                if second_letter == first_letter.upper():
+                    pop_letters_from_inputs(inputs, letter_idx)
+    return len(inputs)
```

Comment:
```suggestion
def get_all_possible_reactions(inputs):
    for letter_idx, first_letter in enumerate(inputs):
        if letter_idx == len(inputs) - 1:
            break
        second_letter = inputs[letter_idx + 1]
        if first_letter.isupper() and second_letter == first_letter.lower():
            pop_letters_from_inputs(inputs, letter_idx)
        elif first_letter.islower() and second_letter == first_letter.upper():
            pop_letters_from_inputs(inputs, letter_idx)
    return len(inputs)
```
Taka mała sugestia zmiany.

1. `inputs[letter_idx + 1]` to zawsze jeden element, więc nie ma sensu po nim iterować, po prostu przypisać go do zmiennej.
2. jak masz 2 ify zagnieżdzone w sobie to możesz je połączyć operatorem logicznym `and`
3. jak masz 2 lub więcej ifów na tym samym poziomie, ale spełnienie jednego wyklucza spełnienie drugiego, to lepiej dać `elif`. Wtedy jak pierwszy `if` zostanie spełniony, to drugi `elif` już nie jest sprawdzany, przez co jest mniej obliczeń.